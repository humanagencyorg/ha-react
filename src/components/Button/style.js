import styled from 'styled-components';
import { darken, lighten } from 'polished';

export const StyledButton = styled.button.attrs((props) => ({
  type: props.type || 'button',
}))`
  width: 100%;
  align-items: center;
  justify-content: center;
  font-size: 1rem;
  font-weight: 600;
  line-height: 2;
  color: #727278;
  padding: 0.5rem;
  background-color: ${(props) =>
    props.backgroundColor ? props.backgroundColor : '#F2F2F2'};
  border: 1px solid transparent;
  border-radius: 0.5rem;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
  transition: background-color 150ms ease;
  overflow: hidden;
  white-space: nowrap;
  display: block;
  text-overflow: ellipsis;

  &:hover {
    color: #727278;
    background-color: ${lighten(0.05, '#F2F2F2')};
  }

  &:focus {
    outline: 0;
  }

  ${(props) =>
    props.responsive &&
    `
    padding: 1rem;
    position: fixed;
    right: 0;
    bottom: 0;
    left: 0;
    border-radius: 0;
    margin: 0;
    padding: 1rem;
    z-index: 1;
    box-shadow: 0 -15px 10px 0 rgba(0, 0, 0, 0.15);


    @media (min-width: 600px) {
      border-radius: 8px;
      box-shadow: none;
      margin: 0.5rem 0 1rem;
      position: static;
      z-index: 0;
    }
  `}
`;

export const StyledWhiteButton = styled(StyledButton)`
  color: #727278;

  &:hover,
  &:hover:disabled {
    color: #727278;
    background-color: ${lighten(0.05, '#F2F2F2')};
  }
`;

export const StyledChatbotLoadingButton = styled(StyledButton)`
  position: relative;
  background-color: #bdbdbd;
  color: #fff;
  opacity: 1 !important;

  &&:hover {
    background-color: ${lighten(0.05, '#BDBDBD')};
    color: #fff;
  }
`;

export const StyledPrimaryButton = styled(StyledButton)`
  background-color: ${(props) =>
    props.backgroundColor ? props.backgroundColor : '#0078F5'};
  color: #fff;

  &&:hover {
    background-color: ${(props) =>
      props.backgroundColor ?
        darken(0.1, props.backgroundColor) :
        darken(0.1, '#0078F5')};
    color: #fff;
  }

  &:disabled {
    background-color: #333;
    color: #bdbdbd;
    cursor: not-allowed;

    &:hover {
      background-color: #242424;
      color: #9c9c9c;
    }
  }
`;

export const StyledSubmissionButton = styled(StyledPrimaryButton)`
  background-color: ${(props) =>
    props.backgroundColor ? props.backgroundColor : '#0078F5'};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transition: transform 0.15s ease;
  margin-bottom: 0;

  &&:hover {
    background-color: ${(props) =>
      props.backgroundColor ?
        darken(0.1, props.backgroundColor) :
        darken(0.1, '#0078F5')};
    color: #fff;
    transform: scale(1.03);
  }
`;

export const StyledNewItemButton = styled(StyledButton)`
  background-color: rgba(53, 53, 63, 0.45);
  color: #fff;
  box-shadow: none;
  justify-content: left;
  border-width: 1px;
  border-style: dashed;
  border-color: rgba(255, 255, 255, 0.6);
  display: flex;
  margin: ${(props) => (props.noMargin ? '0' : '0.5rem 0 1rem 0')};

  &:hover {
    background-color: ${darken(0.1, '#35353f73')};
    color: #fff;
  }
`;

export const StyledSmallDeleteButton = styled.button.attrs(() => ({
  type: 'button',
}))`
  display: flex;
  justify-content: center;
  color: #fff;
  background-color: #ff432e;
  width: 25px;
  height: 25px;
  padding: 0.5rem;
  border: 2px solid #fff;
  border-radius: 50%;
  line-height: 5px;
  cursor: pointer;
  transition: background-color 150ms ease;

  &:hover {
    background-color: ${darken(0.15, '#ff432e')};
  }

  &:focus {
    outline: 0;
  }
`;

export const StyledLinkButton = styled.a`
  -webkit-font-smoothing: antialiased;
  appearance: none;
  background-color: #f2f2f2;
  border-radius: 8px;
  border: 1px solid transparent;
  box-sizing: border-box;
  color: #000;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  font-weight: 600;
  line-height: 2;
  margin: 0.5rem 0 1rem;
  padding: 0.5rem;
  text-align: center;
  text-decoration: none;
  transition: background-color 150ms ease;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  &:hover {
    background-color: ${lighten(0.05, '#F2F2F2')};
    color: #000;
  }

  &:active {
    background-color: #eee;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
    &:hover {
      background-color: #fff;
    }
  }
`;

export const StyledPrimaryLinkButton = styled(StyledLinkButton)`
  background-color: ${(props) =>
    props.backgroundColor ? props.backgroundColor : '#0078F5'};
  color: #fff;

  &&:hover {
    background-color: ${(props) =>
      props.backgroundColor ?
        darken(0.1, props.backgroundColor) :
        darken(0.1, '#0078F5')};
    color: #fff;
  }
`;
