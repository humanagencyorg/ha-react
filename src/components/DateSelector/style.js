const setCustomColor = ({ customColor }) =>
  customColor ? customColor : '#0078f5';

export const styles = {
  root: {
    width: '100%',

    '& .MuiInput-formControl': {
      marginTop: 0,
      backgroundColor: 'transparent',
    },

    '& .MuiFilledInput-root': {
      position: 'relative',
      backgroundColor: 'transparent',
    },

    '& .MuiFormLabel-root.Mui-focused': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
      color: '#727278',
    },

    '& .MuiFormLabel-root.MuiFormLabel-filled': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
    },

    '& .MuiFormLabel-root.Mui-error': {
      color: '#ff432e',
    },

    '& .MuiInputBase-root.Mui-error .MuiInputBase-input': {
      border: '1.5px solid #ff432e',
    },

    '& .MuiInputLabel-filled.MuiInputLabel-shrink': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
    },

    '& .MuiFormHelperText-root.Mui-error': {
      color: '#fff',
    },
  },
  input: {
    background: '#fff',
    padding: '20px 16px 6px',
    borderRadius: '8px',
    border: '1.5px solid',
    borderColor: '#f2f2f2',
    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',

    '&:hover': {
      borderColor: '#f2f2f2',
    },

    '&:disabled': {
      cursor: 'not-allowed',
      backgroundColor: '#e6e6e6',
      '&:hover': {
        border: '1.5px solid',
        borderColor: '#f2f2f2',
      },
    },

    '&:focus': {
      background: '#fff',
      borderColor: setCustomColor,
      boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',
    },
  },
  adornment: {
    position: 'absolute',
    right: '12px',

    '& .MuiIconButton-root': {
      padding: '6px',

      '&:hover': {
        color: '#fff',
        backgroundColor: setCustomColor,
      },
    },

    '&.Mui-error .MuiFilledInput-input': {
      border: '1.5px solid #ff432e',
    },
  },
  dialog: {
    '& .MuiPickersCalendarHeader-iconButton': {
      width: 'auto',
    },
    '& .MuiPickersToolbar-toolbar': {
      display: 'flex',
      height: '96px',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      backgroundColor: setCustomColor,
    },
    '& .MuiPickersDay-daySelected': {
      backgroundColor: setCustomColor,
    },
    '& .MuiPickersToolbarButton-toolbarBtn': {
      width: 'auto',
    },
    '& .MuiButtonBase-root:not(.MuiPickersDay-daySelected)': {
      '&:hover': {
        color: setCustomColor,
      },
    },
    '& + .MuiDialogActions-root .MuiButton-textPrimary': {
      width: 'auto',
      color: setCustomColor,
      '&:hover': {
        color: setCustomColor,
      },
    },
    '& .MuiPickersToolbarText-toolbarTxt': {
      fontSize: '26px',
      lineHeight: '1.235',
      position: 'relative',
      color: '#fff',
    },
    '& h6.MuiPickersToolbarText-toolbarTxt::after': {
      display: 'block',
      content: '""',
      position: 'absolute',
      top: '50%',
      transform: 'translateY(-50%)',
      right: '-24px',
      backgroundImage:
        'url(\'data:image/svg+xml,%3Csvg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="white" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-down"%3E%3Cpolyline points="6 9 12 15 18 9"%3E%3C/polyline%3E%3C/svg%3E\')',
      backgroundSize: 'contain',
      backgroundRepeat: 'no-repeat',
      width: '20px',
      height: '20px',
    },
  },
  label: {
    width: 'calc(100% - 64px)',
    fontSize: '1rem',
    lineHeight: 1.3,
    fontWeight: 500,
    color: '#727278',
    margin: '16px 0 8px',
    transform: 'translate(16px, -1px)',
    zIndex: 1,
    pointerEvents: 'none',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    transition: 'transform 0.2s, font-size 0.2s',

    '&.Mui-error': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
    },
  },
  errorMessage: {
    position: 'absolute',
    top: '-10px',
    right: '6px',
    padding: '2px 8px',
    backgroundColor: '#ff432e',
    borderRadius: '6px',
    zIndex: 1,
  },
  hidden: {
    display: 'none',
  },
};
