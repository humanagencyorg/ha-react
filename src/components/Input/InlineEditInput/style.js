import styled from 'styled-components';

import {
  InputPrefix,
  InputWrapper,
  StyledInputNoLeftRadius,
} from '../style';

export const InlineEditInputPrompt = styled.div`
  background-color: unset;
  color: rgba(255, 255, 255, 0.6);
  padding: 6px 15px 6px 0;
  font-size: 18px;
  font-weight: 500;
  border-width: 0;
  width: 100%;
  margin: 0;
  line-height: 2;
  border-radius: 0 8px 8px 0;
  letter-spacing: -0.5px;
`;

export const Prefix = styled(InputPrefix)`
  color: #000;
  background-color: unset;
  font-size: 18px;
  font-weight: 500;
  width: 2rem;
  margin-left: 8px;
  padding-top: 2px;
`;

export const DisabledPrefix = styled(Prefix)`
  color: rgba(255, 255, 255, 0.8);
`;

export const InputField = styled(StyledInputNoLeftRadius)`
  color: #000;
  background-color: unset;
  padding-left: 0;
  -moz-appearance:textfield;

  &::-webkit-inner-spin-button {
    /* display: none; <- Crashes Chrome on hover */
    -webkit-appearance: none;
    margin: 0; /* <-- Apparently some margin are still there even though it's hidden */
  }

  &::placeholder {
    font-size: 18px;
    font-weight: 500;
    letter-spacing: -0.5px;
  }
`;

export const InlineEditInputWrapper = styled(InputWrapper)`
  margin-bottom: 8px;
  border: 1px solid rgba(255, 255, 255, 0.4);
  box-sizing: border-box;
  border-radius: 8px;
  transform: scale(1);
  transition: transform 0.1s;
  background-color: ${(props) =>
    props.isChecked ? '#fff' : 'rgba(0,0,0,0.2)'};

  &:hover {
    transform: scale(1.01);
    transition: transform 0.1s;
    color: rgba(255, 255, 255, 0.8);
    border: 1px solid #fff;
  }
`;

export const Postfix = styled(Prefix)`
  margin-left: 0;
  margin-right: 12px;
  font-size: 12px;
  text-transform: uppercase;
`;

export const DisabledPostfix = styled(Postfix)`
  color: rgba(255, 255, 255, 0.8);
`;
