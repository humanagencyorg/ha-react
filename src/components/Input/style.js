import styled from 'styled-components';

export const StyledInput = styled.input.attrs((props) => ({
  type: props.type || 'text',
}))`
  background-color: ${props => props.backgroundColor ? props.backgroundColor : '#fff'};
  padding: 0.5rem;
  padding-left: 15px;
  padding-right: 15px;
  border-radius: 8px;
  border-width: 0;
  width: 100%;
  margin: 0;
  box-sizing: border-box;
  line-height: 2;

  &:focus {
    border-width: 0px;
    background-color: #fff;
    box-shadow: none;
    outline: 0;
  }
`;

export const StyledChatbotPhoneInput = styled.input.attrs((props) => ({
  type: props.type || 'text',
}))`
  && {
    background-color: #333;
    padding: 0.5rem;
    padding-left: 1rem;
    padding-right: 1rem;
    border-radius: 8px;
    border-width: 0;
    width: 100%;
    margin: 0;
    color: #FFF;
    flex: 1;
    border-radius: ${props => props.actionButton ? '8px 0 0 8px' : '8px'};

    &:disabled {
      opacity: 1;
    }

    &:focus {
      border-width: 0px;
      box-shadow: none;
    }

    &:hover {
      border-width: 0px;
      box-shadow: none;
    }

    & ~ button {
      flex: 0.4;
      margin: 0;
      border-radius: 0 8px 8px 0;
    }
  }
`;

export const StyledInputNoLeftRadius = styled(StyledInput)`
  border-radius: 0 8px 8px 0;
`;

export const StyledDraggableInput = styled(StyledInput)`
  border-radius: 0 8px 8px 0;
  padding-left: 5px;
`;

export const InputWrapper = styled.div`
  display: flex;
  align-items: stretch;
  position: relative;
  margin: ${ props => props.noMargin ? '0' : '0.5rem 0 1rem 0' };
  width: 100%;
`;

export const StyledErrorMessage = styled.span`
  border-radius: 5px;
  text-align: center;
  background-color: #E74C0A;
  font-size: 12px;
  padding: 6px 12px;
  position: absolute;
  right: 5px;
  top: -5px;
`;

export const InputPrefix = styled.span`
  display: flex;
  align-items: center;
  font-size: 1.5rem;
  justify-content: center;
  width: 3rem;
  background-color: rgba(53, 53, 63, 0.45);
  color: #fff; 
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  line-height: 2;
  
  img {
    display: flex;
  }
`;

export const InputPrefixInline = styled(InputPrefix)`
  background-color: #fff;
  width: 34px;
`;

export const ColorBox = styled.div`
  border-radius: 8px;
  background-color: ${ props => props.color ? props.color : '#0078F5' };
  position: absolute;
  width: 2rem;
  height: 2rem;
  margin-left: 0.5rem;
  margin-top: 0.5rem;
`;

