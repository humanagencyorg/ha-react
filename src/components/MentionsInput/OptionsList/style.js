import styled from 'styled-components';

export const Container = styled.div`
  position: absolute;
  width: 100%;
  height: 0px;
  left: 0;
  z-index: 2;
`;

export const InnerWrapper = styled.div`
  display: flex;
  color: #727278;
  flex-direction: column;
  align-items: center;
  padding-top: 35px;
  padding-bottom: 35px;
`;

export const EmptyResultsText = styled.span`
  max-width: 150px;
  margin-top: 10px;
  font-size: 13px;
  font-weight: 500;
  line-height: 1.5;
  text-align: center;
`;
