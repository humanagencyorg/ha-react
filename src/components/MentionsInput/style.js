import styled from 'styled-components';
import { Textarea } from '../Globals';
import { X, Search } from 'react-feather';

export const StyledTextarea = styled(Textarea)`
  border-radius: 8px 8px 0px 0px;
  padding: 12px 16px;
  box-sizing: border-box;

  &:focus {
    box-shadow: none;
  }

  &::placeholder {
    color: #bdbdbd;
  }
`;

export const ClearIcon = styled(X)`
  position: absolute;
  top: 50%;
  right: 12px;
  transform: translateY(-50%);
  color: #333;
  opacity: 0.6;
  cursor: pointer;

  &:hover {
    opacity: 1;
  }
`;

export const SearchIcon = styled(Search)`
  position: absolute;
  top: 50%;
  right: 12px;
  transform: translateY(-50%);
  color: #727278;
  opacity: 0.6;
  cursor: pointer;
`;
