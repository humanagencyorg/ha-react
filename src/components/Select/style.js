export const componentStyles = {
  root: {
    marginBottom: '8px',

    '& .MuiInput-formControl': {
      marginTop: 0,
      backgroundColor: 'transparent',
    },

    '& .MuiFilledInput-root': {
      position: 'relative',
      backgroundColor: 'transparent',
    },

    '& .MuiFormLabel-root.Mui-focused': {
      color: '#727278',
    },

    '& .MuiFormLabel-root.Mui-error': {
      color: '#ff432e',
    },

    '& .MuiInputBase-root.Mui-error .MuiInputBase-input': {
      border: '1.5px solid #ff432e',
    },

    '& .MuiFormHelperText-root.Mui-error': {
      color: '#fff',
    },
    '& .MuiNativeSelect-select.MuiNativeSelect-select': {
      paddingRight: '44px',
    },
  },
  label: {
    width: 'calc(100% - 60px)',
    fontSize: '12px',
    lineHeight: 1.3,
    fontWeight: 500,
    color: '#727278',
    margin: '16px 0 8px',
    transform: 'translate(16px, -12px)',
    zIndex: 1,
    pointerEvents: 'none',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    transition: 'transform 0.2s, font-size 0.2s',
  },
  errorMessage: {
    position: 'absolute',
    top: '-10px',
    right: '6px',
    padding: '2px 8px',
    backgroundColor: '#ff432e',
    borderRadius: '6px',
    zIndex: 1,
  },
  icon: {
    marginRight: '16px',
  },
  disabled: {
    '&.MuiNativeSelect-select.Mui-disabled': {
      cursor: 'not-allowed',
      backgroundColor: '#e6e6e6',
    },
  },
};

export const inputStyles = {
  input: {
    background: '#fff',
    padding: '20px 16px 6px',
    borderRadius: '8px',
    border: '1.5px solid',
    borderColor: '#fff',
    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',

    '&:hover': {
      borderColor: '#fff',
    },

    '&:focus': {
      background: '#fff',
      borderColor: '#0078f5',
      boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',
      borderRadius: '8px',
    },
  },
};
