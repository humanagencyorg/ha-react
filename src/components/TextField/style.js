export const styles = {
  root: {
    width: '100%',
    '&.MuiTextField-root': {
      marginBottom: '8px',
    },

    '& .MuiInput-formControl': {
      marginTop: 0,
      backgroundColor: 'transparent',
    },

    '& .MuiFilledInput-root': {
      position: 'relative',
      backgroundColor: 'transparent',
    },

    '& .MuiFormLabel-root.Mui-focused': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
      color: '#727278',
    },

    '& .MuiFormLabel-root.MuiFormLabel-filled': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
    },

    '& .MuiFormLabel-root.Mui-error': {
      color: '#ff432e',
    },

    '& .MuiInputBase-root.Mui-error .MuiInputBase-input': {
      border: '1.5px solid #ff432e',
    },

    '& .MuiInputLabel-filled.MuiInputLabel-shrink': {
      transform: 'translate(16px, -12px)',
      fontSize: '12px',
    },

    '& .MuiFormHelperText-root.Mui-error': {
      color: '#fff',
    },
  },
  input: {
    background: '#fff',
    padding: '20px 16px 6px',
    borderRadius: '8px',
    border: '1.5px solid',
    borderColor: '#fff',
    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',

    '&:hover': {
      borderColor: '#fff',
    },

    '&:disabled': {
      '&:hover': {
        border: '1.5px solid',
        borderColor: '#fff',
      },
    },

    '&:focus': {
      background: '#fff',
      borderColor: '#0078f5',
      boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.08)',
    },
  },
  label: {
    fontSize: '1rem',
    lineHeight: 1.3,
    fontWeight: 500,
    margin: '16px 0 8px',
    color: '#727278',
    transform: 'translate(16px, -1px)',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    width: 'calc(100% - 32px)',
    transition: 'transform 0.2s, font-size 0.2s',
  },
  errorMessage: {
    position: 'absolute',
    top: '-10px',
    right: '-8px',
    padding: '2px 8px',
    backgroundColor: '#ff432e',
    borderRadius: '6px',
    zIndex: 1,
  },
};
