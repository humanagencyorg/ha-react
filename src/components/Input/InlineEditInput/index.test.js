import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { InlineEditInput } from './index';
import { InlineEditInputPrompt, Prefix, DisabledPrefix, Postfix, DisabledPostfix } from './style';
import { StyledErrorMessage } from '../style';

describe('InlineEditInput component', () => {
  describe('Other is checked', () => {
    it('matches snapshot', () => {
      const component = renderer.create(<InlineEditInput isChecked={true} />);

      expect(component).toMatchSnapshot();
    });

    it('renders input field', () => {
      const component = mount(<InlineEditInput isChecked={true} />);

      expect(component.find('input').exists()).toBe(true);
    });

    it('renders provided value', () => {
      const amount = 5;
      const component = mount(
        <InlineEditInput isChecked={true} value={amount} onChange={() => {}} />,
      );

      expect(component.find('input').props().value).toBe(amount);
    });

    it('renders provided placeholder', () => {
      const placeholder = 'Enter amount';
      const component = mount(
        <InlineEditInput isChecked={true} placeholder={placeholder} />,
      );

      expect(component.find('input').props().placeholder).toBe(placeholder);
    });

    it('renders prefix with provided icon', () => {
      const iconSymbol = '₴';
      const component = mount(
        <InlineEditInput isChecked={true} icon={iconSymbol} />,
      );

      expect(component.find(Prefix).text()).toBe(iconSymbol);
    });

    it('renders postfix with provided abbreviation', () => {
      const iconAbbreviation = 'gbp';
      const component = mount(
        <InlineEditInput isChecked={true} postfixIcon={iconAbbreviation} />,
      );

      expect(component.find(Postfix).text()).toBe(iconAbbreviation);
    });

    describe('with error', () => {
      it('renders error', () => {
        const errorMessage = 'error message';
        const component = mount(
          <InlineEditInput isChecked={true} error={errorMessage} />,
        );

        const errorComponent = component.find(StyledErrorMessage);

        expect(errorComponent.text()).toBe(errorMessage);
      });
    });
  });

  describe('Other is not checked', () => {
    it('matches snapshot', () => {
      const component = renderer.create(<InlineEditInput isChecked={false} placeholder='placeholder' />);

      expect(component).toMatchSnapshot();
    });

    it('does not render input field', () => {
      const component = mount(<InlineEditInput isChecked={false} />);

      expect(component.find('input').exists()).toBe(false);
    });

    it('renders InlineEditInputPrompt', () => {
      const component = mount(<InlineEditInput isChecked={false} />);

      expect(component.find(InlineEditInputPrompt).exists()).toBe(true);
    });

    it('renders provided placeholder', () => {
      const component = mount(<InlineEditInput isChecked={false} placeholder={'Placeholder'}/>);

      expect(component.find(InlineEditInputPrompt).text()).toBe('Placeholder');
    });

    it('renders prefix with provided icon', () => {
      const iconSymbol = '₴';
      const component = mount(
        <InlineEditInput isChecked={false} icon={iconSymbol} />,
      );

      expect(component.find(DisabledPrefix).text()).toBe(iconSymbol);
    });

    it('renders postfix with provided abbreviation', () => {
      const iconAbbreviation = 'gbp';
      const component = mount(
        <InlineEditInput isChecked={false} postfixIcon={iconAbbreviation} />,
      );

      expect(component.find(DisabledPostfix).text()).toBe(iconAbbreviation);
    });
  });
});
