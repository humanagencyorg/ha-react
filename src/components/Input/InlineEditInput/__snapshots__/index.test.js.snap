// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineEditInput component Other is checked matches snapshot 1`] = `
.c4 {
  background-color: #fff;
  padding: 0.5rem;
  padding-left: 15px;
  padding-right: 15px;
  border-radius: 8px;
  border-width: 0;
  width: 100%;
  margin: 0;
  box-sizing: border-box;
  line-height: 2;
}

.c4:focus {
  border-width: 0px;
  background-color: #fff;
  box-shadow: none;
  outline: 0;
}

.c5 {
  border-radius: 0 8px 8px 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
  margin: 0.5rem 0 1rem 0;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  background-color: rgba(53,53,63,0.45);
  color: #fff;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  line-height: 2;
}

.c2 img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  color: #000;
  background-color: unset;
  font-size: 18px;
  font-weight: 500;
  width: 2rem;
  margin-left: 8px;
  padding-top: 2px;
}

.c6 {
  color: #000;
  background-color: unset;
  padding-left: 0;
  -moz-appearance: textfield;
}

.c6::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c6::-webkit-input-placeholder {
  font-size: 18px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

.c6::-moz-placeholder {
  font-size: 18px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

.c6:-ms-input-placeholder {
  font-size: 18px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

.c6::placeholder {
  font-size: 18px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

.c1 {
  margin-bottom: 8px;
  border: 1px solid rgba(255,255,255,0.4);
  box-sizing: border-box;
  border-radius: 8px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
  background-color: #fff;
}

.c1:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
  color: rgba(255,255,255,0.8);
  border: 1px solid #fff;
}

.c7 {
  margin-left: 0;
  margin-right: 12px;
  font-size: 12px;
  text-transform: uppercase;
}

<div
  className="c0 c1"
>
  <span
    className="c2 c3"
  />
  <input
    className="c4 c5 c6"
    type="text"
  />
  <span
    className="c2 c3 c7"
  />
</div>
`;

exports[`InlineEditInput component Other is not checked matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
  margin: 0.5rem 0 1rem 0;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 3rem;
  background-color: rgba(53,53,63,0.45);
  color: #fff;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  line-height: 2;
}

.c2 img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: unset;
  color: rgba(255,255,255,0.6);
  padding: 6px 15px 6px 0;
  font-size: 18px;
  font-weight: 500;
  border-width: 0;
  width: 100%;
  margin: 0;
  line-height: 2;
  border-radius: 0 8px 8px 0;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

.c3 {
  color: #000;
  background-color: unset;
  font-size: 18px;
  font-weight: 500;
  width: 2rem;
  margin-left: 8px;
  padding-top: 2px;
}

.c4 {
  color: rgba(255,255,255,0.8);
}

.c1 {
  margin-bottom: 8px;
  border: 1px solid rgba(255,255,255,0.4);
  box-sizing: border-box;
  border-radius: 8px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
  background-color: rgba(0,0,0,0.2);
}

.c1:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  -webkit-transition: -webkit-transform 0.1s;
  -webkit-transition: transform 0.1s;
  transition: transform 0.1s;
  color: rgba(255,255,255,0.8);
  border: 1px solid #fff;
}

.c6 {
  margin-left: 0;
  margin-right: 12px;
  font-size: 12px;
  text-transform: uppercase;
}

.c7 {
  color: rgba(255,255,255,0.8);
}

<div
  className="c0 c1"
>
  <span
    className="c2 c3 c4"
  />
  <div
    className="c5"
    placeholder="placeholder"
  >
    placeholder
  </div>
  <span
    className="c2 c3 c6 c7"
  />
</div>
`;
